autopilot/contractor.go

```diff
@@ -1163,19 +1164,24 @@ func (c *contractor) formContract(ctx context.Context, w Worker, host hostdb.Hos
 	return formedContract, true, nil
 }
 
-func (c *contractor) priceTable(ctx context.Context, w Worker, hk types.PublicKey, siamuxAddr string) (*hostdb.HostPriceTable, error) {
-	ctx, cancel := context.WithTimeout(ctx, contractHostPriceTableTimeout)
-	defer cancel()
-
-	pt, err := w.RHPPriceTable(ctx, hk, siamuxAddr)
-	if err != nil {
-		return nil, err
+func (c *contractor) priceTable(ctx context.Context, w Worker, host hostdb.Host) (hostdb.HostPriceTable, error) {
+	// fetch the settings if necessary
+	if !host.Scanned || host.Settings.NetAddress == "" {
```

Comment:
The `host.Settings.NetAddress == ""` check will go away once I've added the extra settings validation when we scan a host.